### DISCUSSION TITLE
How to disable console.log?

### DISCUSSION THREAD

**Discussion by mike35x95x1** [requarks/wiki#3198] [Help / Questions] [answered]
Wiki.js creates lot of messages in the console (console.log) which is common issue when the node becomes slow. There are no benchmarks to confirm that my wiki.js is slower than it should normally be so I am trying to eliminate common issues.

Is it possible to disable the console.log? Keeping the logs in a database or file would be a benefit of course.

OS: Ubuntu (bionic) 18.04 x64
Wiki: 2.5.170
Database: PostgreSQL 13.2

**Comment by mattpr** [comment, marked as answer]
Howdy.

### 1) Reduce log verbosity
You can set log level in config.yml
`logLevel: info`
Reducing this will reduce output.

### 2) Output isn't necessarily writing to files

https://nodejs.org/api/console.html#console_console

```javascript
console.log('hello %s', 'world');
// Prints: hello world, to stdout
console.error(new Error('Whoops, something bad happened'));
// Prints error message and stack trace to stderr:
```

console.log/error is writing to stdout/error.  When you run node app directly in console, stdout/err is written to console (slow).  If you redirect output to a file it is writing to a file without optimized buffering (slow).

So yes, if you are running wiki.js directly from the console it will output to console which is slow.

In normal production environments you should be running this with a process manager (systemd or pm2 if you are not doing devops or only run node stuff)...in those cases systemd (journal) or pm2 logging will manage the logging (consumption of stdout/err from the process) and should both be smart about buffering output and not blocking.

### 3) this is a wiki, not a high-volume messaging service

The number of qps we are talking about should not be getting into the territory where we are worried about logging...unless you are dealing with a giant wiki with a huge number of simultaneous users...in which case you will have other bigger problems than whether logging to the journal is slowing down your wiki.

**Comment by NGPixel** [comment]
As @mattpr mentionned, it is **VERY unlikely** that logging is the cause of any slowdown.
